show/main.py

```diff
@@ -888,23 +892,213 @@ def get_bgp_peer():
+        # jason option and then filter out the json entries (by removing those next Hop that are 
+        # back-end ASIC if display is for front-end only). If the entry itself has no more next Hop after filtering, 
+        # then skip over that particular route entry.  Once completed, if the user chose "json" option, 
+        # then just print out the dictionary in Json format accordingly. But if no "json" option specified,
+        # then print out the header and the decoded entry representation for each route accordingly.
+        # This code is based on FRR 7.2 branch. If we moved to a new version we may need to change here as well
+        asic_cnt = multi_asic.get_num_asics()
+        arg_strg = ""
+        found_json = 0
+        back_end_intf_str = "PortChannel4"
+        if display == "all":
+            filter_back_end = False
+        else:
+            filter_back_end = True
+        if str(namespace) == "None":
+            start_range = 0
+            end_range = asic_cnt
+        else:
+            # -n asic0|asic1|...|asicN    So skip the first 4 chars to get the namespace number only
+            start_range = int(namespace[4:])
+            end_range = start_range + 1
+        hdr_printed = False
+        del_cnt = 0
+        # get all the other arguments except json that needs to be the last argument of the cmd if present
+        for arg in args:
+            arg_strg += " " + str(arg)
+            if str(arg) == "json":
+                found_json = 1
+        if not found_json:
+            arg_strg += " json"
+        for ns in range(start_range, end_range):
+            filtered_route = {}
+            # Need to add "ns" to form bgpX so it is sent to the correct bgpX docker to handle the request
+            cmd = 'sudo docker exec bgp' + str(ns) + ' vtysh -c "show ip route' + arg_strg + '"'
+            output = commands.getoutput(cmd)
+            # in case no output or something went wrong with user specified cmd argument(s) error it out
+            # error from FRR always start with character "%"
+            if output == "":
+                return
+            if output[0] == "%":
+                # remove the "json" keyword that was added by this handler so it wont confuse the user
+                error_msg = output[:-4]
+                print error_msg
+                return
+            route_info = json.loads(output)
+            if filter_back_end:
+                # clean up the dictionary to remove all the nexthops that are back-end interface "PortChannel4xxx"
+                for route, info in route_info.items():
+                    new_info_l = []
+                    new_info_cnt = 0
+                    for i in range(0, len(info)):
+                        new_info = info[i]
+                        new_nhop_l = []
+                        del_cnt = 0
+                        while len(info[i]['nexthops']):
```

Comment:
Can we use these function to check if the interface is internal instead of hardcording the interface name
[is_port_internal](https://github.com/Azure/sonic-buildimage/blob/a8016a6987e4225ef9f35bae0148f07f4e0f7865/src/sonic-py-common/sonic_py_common/multi_asic.py#L251) 
[is_port_channel_internal](https://github.com/Azure/sonic-buildimage/blob/a8016a6987e4225ef9f35bae0148f07f4e0f7865/src/sonic-py-common/sonic_py_common/multi_asic.py#L272)

Reply:
Agreed. Removed hardcoded interface name and only use skip_display to decide if display shall be skipped.